packages/appstore/src/components/modals/wallet-modal/wallet-modal.tsx

```diff
@@ -1,19 +1,44 @@
 import React from 'react';
 import { Modal } from '@deriv/components';
+import { useCurrencyConfig, useWalletList } from '@deriv/hooks';
+import { observer, useStore } from '@deriv/stores';
+import { localize } from '@deriv/translations';
 import WalletModalHeader from './wallet-modal-header';
 import WalletModalBody from './wallet-modal-body';
-import { observer, useStore } from '@deriv/stores';
+import type { TWalletType } from './provider';
+
+// TODO: remove this type when we can use real current wallet object
+export type TWallet = Omit<Exclude<ReturnType<typeof useWalletList>['data'], undefined>[0], 'balance'> & {
+    balance?: string | number;
+    is_crypto?: boolean;
+    is_demo: boolean;
+    name: string;
+    wallet_type: TWalletType;
+};
 
 const WalletModal = observer(() => {
     const store = useStore();
     const {
-        client: { balance, currency, landing_company_shortcode: shortcode },
+        client: { balance, currency, landing_company_shortcode },
         ui: { is_dark_mode_on, is_wallet_modal_visible, is_mobile, setIsWalletModalVisible },
         traders_hub: { is_demo },
     } = store;
 
-    // TODO: Temporary wallet type. Will be refactored later. Add correct type
-    const wallet_type = is_demo ? 'demo' : 'real';
+    const { getConfig } = useCurrencyConfig();
+    const currency_config = getConfig(currency);
+    const is_crypto = currency_config?.is_crypto;
+    const currency_display_code = currency_config?.display_code;
+
+    // TODO: Replace this object with current wallet
+    const wallet = {
+        balance,
+        currency,
+        is_crypto,
+        is_demo,
+        landing_company_shortcode,
+        name: `${is_demo ? localize('Demo') : ''} ${currency_display_code} ${localize('Wallet')}`,
+        wallet_type: (is_demo ? 'demo' : 'real') as TWalletType,
+    };
```

Comment:
@heorhi-deriv Why do you need to create this object? Can't we just use the `wallet` object itself? 🤔